src/typeahead/dataset.js

```diff
@@ -261,20 +261,21 @@ var Dataset = (function() {
           that.trigger('asyncRequested', query);
         }
       }
-
+      
       function async(suggestions) {
         suggestions = suggestions || [];
-
+        
         // if the update has been canceled or if the query has changed
         // do not render the suggestions as they've become outdated
         if (!canceled && rendered < that.limit) {
-          that.cancel = $.noop;
-          that._append(query, suggestions.slice(0, that.limit - rendered));
-          rendered += suggestions.length;
-
-          that.async && that.trigger('asyncReceived', query);
+            that.cancel = $.noop;
+            var idx = Math.abs(rendered - that.limit);
+            rendered += idx;
+            that._append(query, suggestions.slice(0, idx));
```

Comment:
Since the limit is 5 by default and my suggestions is rendering 5 items, wouldn't this return empty suggestions?

Example:

```javascript
> Math.abs(5 - 5)
0
> [1, 2, 3].slice(0, 0)
[] (0)
```